src/platform/arguments.c

```diff
@@ -0,0 +1,95 @@
+#include "arguments.h"
+
+#include "SDL.h"
+
+#define CURSOR_SCALE_ERROR_MESSAGE "Option --cursor-scale must be followed by a scale value of 1, 1.5 or 2"
+#define DISPLAY_SCALE_ERROR_MESSAGE "Option --display-scale must be followed by a scale value between 0.5 and 5"
+
+static int parse_decimal_as_percentage(const char *str)
+{
+    const char *start = str;
+    char *end;
+    long whole = SDL_strtol(start, &end, 10);
+    int percentage = 100 * (int) whole;
+    if (*end == ',' || *end == '.') {
+        end++;
+        start = end;
+        long fraction = SDL_strtol(start, &end, 10);
+        switch (end - start) {
+            case 0:
+                break;
+            case 1:
+                percentage += fraction * 10;
+                break;
+            case 2:
+                percentage += fraction;
+                break;
+            default: {
+                int fraction_digits = end - start;
+                while (fraction_digits > 2) {
+                    fraction = fraction / 10;
+                    fraction_digits--;
+                }
+                percentage += fraction;
+                break;
+            }
+        }
+    }
+    if (*end) {
+        // still some characters left, print out warning
+        SDL_Log("Invalid decimal: %s\n", str);
+        return -1;
+    }
+    return percentage;
+}
+
+int platform_parse_arguments(int argc, char **argv, julius_args *output_args)
+{
+    int ok = 1;
+
+    // Set sensible defaults
+    output_args->data_directory = 0;
+    output_args->display_scale_percentage = 100;
+    output_args->display_scale_percentage = 100;
```

Comment:
this should probably be output_args->cursor_scale_percentage?

Reply:
Yeah, if this is not properly initialized the cursor is always loaded at 2x scale for me without setting anything on the command line. It's looking great on my surface, but probably wrong for most people!

Reply:
Whoops, copy/paste error..